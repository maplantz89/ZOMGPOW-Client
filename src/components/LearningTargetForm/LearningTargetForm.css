 .learning-target-form {
  display: flex; 
  flex-direction: column;
  padding: 15px;
  border-radius: 15px;
  margin: 40px auto;
  width: 80%;
  max-width: 300px;
  align-items: center;
 }

 .learning-target-form label {
   font-size: 25px;
   font-weight: 800;
 }

 .learning-target-form textarea {
  border-radius: 5px;
  font-family: 'Montserrat', sans-serif;
  font-size: 15px;
  width: 300px;
  height: 50px;
  border: solid 1px darkgray;
  margin: 10px;
 }

 .learning-target-form button {
  color: white;
  background: #81D40A;
  padding: 5px 15px 8px 15px;
  border-radius: 15px;
  font-size: 16px;
  text-align: center;
  font-family: 'Lexend Deca', sans-serif;
 }

 .learning-target-form button:hover {
  background: black;
  cursor: pointer;
 }