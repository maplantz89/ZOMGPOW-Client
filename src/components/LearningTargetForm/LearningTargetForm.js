import React from 'react';
import ExitTicketForm from '../ExitTicketForm/ExitTicketForm';
import './LearningTargetForm.css';

class LearningTargetForm extends React.Component {

  state = {
    error: null,
  }
  staticDefaultProps={}

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.showModal();
  }
  render() {
    return (
      <div className='learning-target-container'>
      <ExitTicketForm updateExitTicket={this.updateExitTicket}/>
      <form className='learning-target-form'
        onSubmit={(e) => this.handleSubmit(e)}>
            <label htmlFor='learningTarget'>Learning Target:</label>
            <textarea 
              id='learning-target'
              onChange={this.props.handleChange}
              value={this.props.learningTarget}
              name='learningTarget'
              aria-label='learning target'
              aria-required='true'
              required
              />
            <div>
              <button type='submit' className='button green-button'>Start Session</button>
            </div>
          </form>
          </div>
    )
  }
}

export default LearningTargetForm