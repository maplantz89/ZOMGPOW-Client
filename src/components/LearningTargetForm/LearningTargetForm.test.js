import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LearningTargetForm from './LearningTargetForm';

describe(`LearningTargetForm component`, () => {
  it('renders a .LearningTargetForm by default', () => {
    const wrapper = shallow(<LearningTargetForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})