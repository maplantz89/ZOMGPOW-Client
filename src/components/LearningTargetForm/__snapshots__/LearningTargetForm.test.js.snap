// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningTargetForm component renders a .LearningTargetForm by default 1`] = `
<form>
  <label
    htmlFor="learning-target"
  >
    Learning Target:
  </label>
  <input
    aria-label="learning target"
    aria-required="true"
    id="learning-target"
    name="learning-target"
    onChange={[Function]}
    required={true}
    value=""
  />
  <div>
    <button
      type="submit"
    >
      Start Session
    </button>
  </div>
</form>
`;
